### PR TITLE
disable clean db option

### PR DESCRIPTION
**Pull request by aldarund**
With beehive every worker run a clean db loop in a separate thread. It increase opened connections to db  twice. Also it cause a lot of deadlock issues when multiple workers run multiple db clean loops. And its not really needed. One db clean loop for one database more than enough.
So to disable it i added option --disable-clean.

### CODE REVIEW

**Comment by gabrielsoldani** [comment]
+1. This fixes issue #1034.

**Comment by antiremy** [comment]
If we only need it for one thread, why not make it an --enable-clean argument? That way you only need to add it to one command versus adding --disable-clean to a bunch

**Comment by kbtbc** [comment]
Perhaps enable db clean by default, esp with the -os flag, since in beehive it usually has no workers.  Any -ns threads, disabled by default.  (Do you even need an extra flag?)

**Comment by gabrielsoldani** [comment]
You do, because you can run more than one --only-server instance for load
balancing, and you'd still want only one of them to do the clean up
procedures.

In my opinion, a disable flag is the best option right now. If we ever
split the web server from the search we could more intelligently handle
this. In the best case scenario the webserver would only ever need SELECT
permissions.

**Comment by aldarund** [comment]
I will change pr to following.
1) In instances with -ns clean will be disabled.
2) In instances where server is run it will be enabled by default and --disable-clean will disable it.
Any objections?

**Comment by gabrielsoldani** [comment]
I disagree. People might not want to run the web server (for data
gathering, web hooks, different front-end, etc).

Make it the other way around (--only-server never cleans up, otherwise
clean unless --disable-clean is set). The web server should not handle
clean up, this is not a task for the front-end.

**Comment by aldarund** [comment]
The other way around will mean adding this flag to every search worker, which is very verbose.

**Comment by aldarund** [comment]
Usual setup is when  there 1 server with multiple workers, or a serve with a worker. When there no server at all is quite uncommon.

**Comment by gabrielsoldani** [comment]
Uncommon but possible (in fact it's my setup, since I use my own front-end).

This change (where --no-server has no clean up by default with no way to
enable) would make me and other users run a web server just for clean up,
which makes no sense.

Let me rephrase that: **having our web servers, which are supposed to serve
web pages and data, to clean up after our workers, is a bad design
decision. Having them being the _only_ component which can clean up after
our workers is a _terrible_ design decision.**

Right now we can have many web servers (including zero) and many searchers
(including zero). Just keep the commit exactly like it is right now
(everything runs the clean up unless --disable-clean is set) and it's an
upgrade. If we require a webserver to do clean ups, it's a downgrade.

**Comment by aldarund** [comment]
Well, then best option would be to have cleanup disabled by default, and need to be enabled with flag where its needed.

**Comment by justsomedudeonthenet** [comment]
My opinion:

- Disabled by default is a **bad** idea.  People won't enable it.  Then they will run into issues.
- Enabled only for -os is also kind of bad.  I'd rather the webserver instance focus on web serving.  Some people run multiple -os, which defeats the purpose of this.  Some people run none and just use webhooks.

My vote would be to have it enabled by default on all instances, with the switch to disable it.  It doesn't break any existing behaviour, and people who know what they are doing can make their own decisions on which instances should run the cleanup.

So +1 merge as is please!
